### DISCUSSION TITLE
Is GitHub Actions going to be killed soon?

### DISCUSSION THREAD

**Discussion by ericoporto** [community/community#26312] [Actions]
<p>So I was looking into a workflow that stopped working and noticed that the problem was at uploading the release. No problem let’s see how the release upload works…Oops, GitHub has archived the release asset upload action. And the release create action… And the download is also practically unmaintained… There’s no currently actions for these provided by the community.</p>
<p>Should I just migrate my stuff back to Azure Pipelines? Switch entirely to GitLab?</p>
<p>What’s going on with GitHub Actions development?</p>

**Comment by mickeygousset** [comment]
<p>GitHub Actions is still going strong, in my opinion. You can look at the public roadmap and see what all we have delivered and what is coming in the future.</p>
<p>That said, I can’t speak to why they might have archived those particular action repositories.  Using APIs, you could still create releases directly in your workflows, or create a custom action to do them.  Which I know is extra work.  But it is an option.</p>

**Comment by ericoporto** [comment]
<p>Did you notice how no one uses GitHub Actions seriously? It’s just used in toy workflows and doesn’t do anything real.</p>

**Comment by mickeygousset** [comment]
<p>Hey <a class="mention" href="/ericoporto">@ericoporto</a>! Always up for a lively debate.  I was wondering if you could provide me some details on why you think GitHub Actions isn’t taken seriously and doesn’t do anything real.  It is currently the <span class="hashtag">#1</span> CI system used on GitHub, and I can personally attest to working with many customers to implement GitHub Actions, both from a CI/CD perspective, as well as a general workflow.  That is one of the strengths of GitHub Actions, is the ability to trigger off of any event that GitHub can throw, which allows you to automate beyond just CI/CD?</p>
<p>Can you provide me some details?</p>

**Comment by ericoporto** [comment]
<p>No dogfooding from either GitHub or Microsoft, navigate their repos and you will see they simply don’t use it. Basically no one ships releases from it, people only runs basic tests in it. Look any big cross platform project and see it’s never used.</p>
<p>Anyone thinking serious should just stay on Azure, it’s better supported, has possibility to integrate with manual workflows using Azure Classic together. It’s just works good, support is good, docs is good. I don’t see actions ever competing at this pace of development.</p>

**Comment by mickeygousset** [comment]
<p>Hey <a class="mention" href="/ericoporto">@ericoporto</a> Thanks for the feedback.  I’ll also qualify that everything I say is my personal opinion. :slight_smile:</p>
<p>I’m going to push back on you about the “dogfooding” statement.  Internally at GitHub, we make extensive use of GitHub Actions, for a variety of workflows.  My team itself has workflows that we use to help automate what we do.  I don’t think just browsing the GitHub/Microsoft public repos gives you enough information to make a general statement that GitHub/Microsoft aren’t using actions.</p>
<p>I’m also going to dispute your claim of it is not being used by large projects.  Here are some examples:</p>
<p><a href="https://github.com/microsoft/vscode" class="inline-onebox">GitHub - microsoft/vscode: Visual Studio Code</a> - 116k stars<br>
<a href="https://github.com/rust-lang/rust" class="inline-onebox">GitHub - rust-lang/rust: Empowering everyone to build reliable and efficient software.</a> - 55k stars<br>
<a href="https://github.com/home-assistant/core" class="inline-onebox">GitHub - home-assistant/core: Open source home automation that puts local control and privacy first</a> - 43k stars</p>
<p>Azure Pipelines is a mature product with a lot of users. And I have an extensive background with it, having worked with it since its initial version as Visual Studio Team System back in 2005 (yes I’m old). There are a couple of places where Azure Pipelines has a slight edge on Actions, but if you look at our public roadmap, you’ll see that those gaps are closing quickly.</p>

**Comment by ericoporto** [comment]
<p>0 of these use a community maintained action and 0 of these have a release pipeline.</p>
<p>Sorry, but this doesn’t give a good image of the GH Actions product.</p>
<p>It’s good that you are from GH, maybe you can push some internal change.</p>
<p>In my opinion creating a release and pushing assets to it are basic features that have to be maintained by people from GitHub since it has a close relationship with GitHub API. I don’t want to trust my GitHub Token to a third party!</p>
<p>We can only use JS Actions for cross platform support. What about Powershell, Bash, Python, why they have to be restricted to only Linux? If project is not natively a Typescript/Javascript it’s hard to switch tools to maintain your own actions for your workflow. I find the approach from CirrusCi has some good elements that is worth taking a look - it looks aligned to what Actions is working towards.</p>
<p>Other problem is there’s no GitHub Actions community right now, everything available is too immature. We need better reference documentation for it! Video teaching materials, a better way to search for stuff in each other’s CI workflows.</p>
<p>Sorry for being very critical, but GitHub sets a really high bar for everything, which is good, you guys do good work. But GitHub Actions really need more work before it can shine. After using Jenkins, GitLab, Travis, Circle CI, AppVeyor, TS, Azure, and CirrusCi I expected to settle for Actions and was a bit disappointed that it had to be another one for me to juggle.</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>Hey <a class="mention" href="/ericoporto">@ericoporto</a>! All good feedback thank you. I’m going to reach out to the Actions team, to see if there is any information that can be shared on why those actions were archived. I’ll report back here if I find out anything.</p>
<p>I do feel the GitHub Actions documentation is pretty solid, IMO. :slight_smile:</p>
<p>And don’t ever apologize about being critical.  We have a growth mindset here at GitHub, so we are all about constantly learning, and feedback from our customers is the primary way we can do that.</p>

**Comment by jsoref** [comment]
<p>fwiw, I know of a bunch of Microsoft repositories that use my <a href="https://github.com/marketplace/actions/check-spelling">check-spelling</a> action, including:</p>
<ul>
<li><a href="https://github.com/microsoft/PowerToys/blob/3548e6820efd4baf52526dbcc7b3ee1cbab5fcd2/.github/workflows/spelling2.yml">PowerToys</a></li>
<li><a href="https://github.com/microsoft/Terminal/blob/6ffc3dc7a8a58c9961625618484cab431954e999/.github/workflows/spelling2.yml">Terminal</a></li>
</ul>
<hr>
<p>As an individual maintaining an action, I understand the tension between releasing an update and making sure that it doesn’t have any regressions.<br>
(I’m still trying to figure out a few edges before I make a new release.)</p>
<hr>
<p>That said, I am sympathetic to complaints about the state of the state of repositories in the</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/actions">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/actions" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181096829-29993f42-c05b-4a47-97a1-865d138b0878.png" class="thumbnail onebox-avatar" width="280" height="280">

<h3><a href="https://github.com/actions" target="_blank" rel="noopener">GitHub Actions</a></h3>

  <p>Automate your GitHub workflows. GitHub Actions has 47 repositories available. Follow their code on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
organization.</p>
<p>If a GitHub staff member wants to reach out to me, I’d be happy to discuss some of the problems (although they’re pretty easy to identify).</p>

**Comment by ericoporto** [comment]
<p>A lot has matured in GitHub Actions since my initial post here, I haven’t actually switched fully to GitHub Actions, still juggling between GA, Cirrus CI, Circle CI and AppVeyor, but at least I have been able to use GitHub Actions in more complex and useful workflows now.</p>
